File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.js
function solution(data, col, row_begin, row_end) {
    data = data.sort((a, b) => {
        if(a[col-1] === b[col-1]) return b[0] - a[0];
        return a[col-1] - b[col-1];
    });

    const result = [];
    for(let i=row_begin; i<=row_end; i++) {
        const row = data[i-1];
        const value = row.reduce((a, c) => {
            return a + (c % i);
        }, 0);

        result.push(value);
    }

    return result.reduce((a, c) => {
        if(a === null) return c;
        return a^c;
    }, null);
}

File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.py
from functools import reduce

def solution(data, col, row_begin, row_end):
    data.sort(key = lambda x : (x[col-1], -x[0]))
    return reduce(lambda x, y: x ^ y,
                  [sum(map(lambda x: x%(i+1), data[i])) for i in range(row_begin-1, row_end)])